ckanext/activity/logic/action.py

```diff
@@ -246,6 +247,19 @@ def package_activity_list(
     after = data_dict.get("after")
     before = data_dict.get("before")
 
+    # Get activity permission labels
+    labels = get_permission_labels()
+    activities = model.Session.query(
+        model_activity.Activity).filter_by(object_id=package.id)
+    for activity in activities:
+        activity.permission_labels = labels.get_activity_labels(activity)
+    model.Session.commit()
```

Comment:
updating the activity labels from a "side_effect_free" action doesn't make sense. Shouldn't these have already been set when the activities were created?

Reply:
@wardi I've moved the initial set of the activity labels to when the activity is created. But aren't there things (ex. changing dataset's organization) that would require a change in the activity labels? How should that happen?

Reply:
If we leave the labels on the activities as they are then members of the old org can see changes that occurred while the dataset was part of that org and members of the new org can only see changes that occurred since. I think that's a sensible default.